gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/ExploreService.java

```diff
@@ -103,14 +109,18 @@ public ExploreResponse explore(RequestContext requestContext, ExploreRequest req
     }
   }
 
-  private boolean isContextAnEntityType(ExploreRequest request) {
-    return Arrays.stream(EntityType.values())
-        .anyMatch(entityType -> entityType.name().equalsIgnoreCase(request.getContext()));
+  private boolean isContextAnEntityType(ExploreRequest request, String tenantId) {
+    return Stream.concat(
+            entityTypesProvider.getEntityTypes(tenantId).stream(),
```

Comment:
wouldnt entity types provider will provide the complete list of types? do we need stream of enums?

Reply:
It should. @skjindal93 is it safe to remove the dependency on the enums?